File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    length = len(nums) / 2 # 고를 수 있는 폰켓몬 수
    p_type = set(nums) # 중복 제거한 폰켓몬 종류
    
    for v in p_type : # 폰켓몬 종류만큼 반복
        if(answer < length): # 최대한 데려갈 수 있는 종류 수 < 고를 수 있는 폰켓몬 수
            answer += 1
    
    return answer

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {

        int pick = nums.length / 2; // 선택할 수 있는 포켓몬의 갯수
        
        HashSet<Integer> set = new HashSet<>();
        for(int n : nums){ // 해시셋에 모든 포켓몬 추가(중복은 들어가지 않음)
            set.add(n);
        }
        
        // (선택할 수 있는 폰켓몬의 수 >= 폰켓몬 종류의 수) 라면
        // 답은 폰캣몬 종류의 수
        if(pick >= set.size()) 
            return set.size();
        else // 아니라면 선택할 수 있는 폰켓몬의 수
            return pick;
    }
}